File: integrations/aws-strands/error-code-corpus/typescript/spread_frame.ts
/** A frame built by spreading a base frame and overriding its fields. */
import { EventType, type BaseEvent } from "@ag-ui/core";

const ERROR_BASE = {
  type: EventType.RUN_ERROR,
  message: "Base message.",
  code: "SPREAD_BASE",
};

export function emit(): BaseEvent {
  return {
    ...ERROR_BASE,
    message: "A spread-built message.",
    code: "SPREAD_FRAME",
  };
}

File: integrations/aws-strands/error-code-corpus/python/spread_frame.py
"""A frame built by unpacking a base frame and overriding its fields."""

from ag_ui.core import EventType, RunErrorEvent

ERROR_BASE = RunErrorEvent(
    type=EventType.RUN_ERROR,
    message="Base message.",
    code="SPREAD_BASE",
)


def emit() -> RunErrorEvent:
    return RunErrorEvent(
        **ERROR_BASE.model_dump(exclude={"message", "code"}),
        message="A spread-built message.",
        code="SPREAD_FRAME",
    )
